### ISSUE TITLE
False negative with SIM108

### ISSUE THREAD

**Issue by NeilGirdhar** [astral-sh/ruff#9512]
```python
def main() -> None:
    with open('a'):
        for i in range(10):
            yrange = None
            integer_yrange: None | range
            if yrange is not None:
                integer_yrange = range(int(yrange[0]), int(yrange[1]) + 1)
            else:
                integer_yrange = None
            print(integer_yrange)
```
Does not produce use-ternary-operator.  Removing the function, context manager and loop does.

**Comment by charliermarsh** [comment]
I believe this is https://github.com/astral-sh/ruff/issues/8106 -- it does flag it if you increase the line length: https://play.ruff.rs/c11cd4e9-a96c-40e6-ae60-6c735ab5bfb0

**Comment by NeilGirdhar** [comment]
Sorry, but why does line length matter here?  I think the ternary expression is still an improvement even if it doesn't fit on one line.  Can we format it like so:
```python
                if not isinstance(inference_result, RivalConfiguration):
                    code = None
                else:
                    code = np.asarray(inference_result.pooling_output, dtype=np.float64)
```
becomes
```python
runner_image = (something[0].url 
                if something is not None and len(something) > 1 
                else None)
```

**Comment by charliermarsh** [comment]
(You should contribute to the discussion in the linked issue, which is open.)

**Comment by NeilGirdhar** [comment]
Sorry, my mistake :)
